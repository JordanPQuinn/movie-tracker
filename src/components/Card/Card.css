h1 {
  font-family: 'Anton', sans-serif;
}

h2 {
  font-family: 'Anton', sans-serif;
  font-size: 32px;
  margin: 0;
}

h3, p {
  margin: 0;
  font-family: 'Lato', sans-serif;
}

h3 {
  margin-bottom: 5px;
}

.warning {
  position: absolute;
  background-color: rgba(0, 0, 0, .8);
  color: white;
  height: 100%;
  width:100%;
}

.warning h3 {
  margin-top: 140px;
}

.Card {
  position: relative;
  margin: 0 auto 20px auto;
  width: 300px;
  box-sizing: border-box;
  background-color: #f1f1f1;
  border: 2px solid #cccccc;
}

.favorite-img {
  position: absolute;
  top: 6px;
  left: 8px;
  width: 50px;
  height: 50px;
  background: none;
  border: none;
  background-image: url('../../images/favorite.svg');
}

.favorite-img:hover {
  background-image: url('../../images/favorite-hover.svg');
  cursor: pointer;
}

.information {
  box-sizing: border-box;
  width: 300px;
  padding: 7px;
}

.poster {
  width: 297px; 
}

@media screen and (min-width: 545px) {
  .Card {
    display: flex;
    width: 490px;
    height: 350px;
  }

  .information {
    height: 329px;
    overflow-y: scroll;
  }

  .poster {
    width: 250px;
  }
}