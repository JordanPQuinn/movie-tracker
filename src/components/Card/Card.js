import React from 'react';

export const Card = ({ poster, title, overview, date, rating, id}) => {
  return (
    <article className='Card'>
      <img src={`https://image.tmdb.org/t/p/w500${poster}`} alt="movie-poster" />
      <h2>{ title }</h2>
      <h3>{ date }</h3>
      <p>{ overview }</p>
    </article>
  )
}