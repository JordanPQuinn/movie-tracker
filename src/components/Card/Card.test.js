import React from 'react';
import { shallow } from 'enzyme';
import { Card } from './Card';

//toggle favorite called on button click
//toggle favorite calls determine warning if clicked is false
//toggle favorite chnanges state if clicked is true
//when detemrine warning is called and props.user.name does not exist, set state to true
//when determine warning is called and props.user.name is true, calls handle favorite
//state defaults to clicked: false on load

describe('Card', () => {
  let renderedCard;
  beforeEach(() => {
    const mockFilm = {
      poster: '',
      title: '',
      overview: '',
      date: '',
      rating: 5,
      id: 1
    };

    renderedCard = shallow(<Card film={mockFilm} />);
  });

  it('should match the snapshot', () => {
    expect(renderedCard).toMatchSnapshot();
  });

  //eslint-disable-next-line max-len
  xit('should call toggleFavorite on click with the targeted film object', () => {});

  xit('should change UI to indicate its favorite status', () => {
    expect(renderedCard).toMatchSnapshot();
    //call toggleFavorite
    //expect(renderedCard).toMatchSnapshot();
  });
});
