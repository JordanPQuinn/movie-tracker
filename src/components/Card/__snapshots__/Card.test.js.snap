// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    film={
        Object {
            "date": "",
            "id": 1,
            "overview": "",
            "poster": "",
            "rating": 5,
            "title": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="favorite-img"
          id={undefined}
          onClick={[Function]}
/>,
        <img
          alt="movie-poster"
          className="poster"
          src="https://image.tmdb.org/t/p/w500undefined"
/>,
        <div />,
        <div
          className="information"
>
          <h2 />
          <h3 />
          <p />
</div>,
      ],
      "className": "Card",
      "id": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "favorite-img",
          "id": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "movie-poster",
          "className": "poster",
          "src": "https://image.tmdb.org/t/p/w500undefined",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2 />,
            <h3 />,
            <p />,
          ],
          "className": "information",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="favorite-img"
            id={undefined}
            onClick={[Function]}
/>,
          <img
            alt="movie-poster"
            className="poster"
            src="https://image.tmdb.org/t/p/w500undefined"
/>,
          <div />,
          <div
            className="information"
>
            <h2 />
            <h3 />
            <p />
</div>,
        ],
        "className": "Card",
        "id": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "favorite-img",
            "id": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "movie-poster",
            "className": "poster",
            "src": "https://image.tmdb.org/t/p/w500undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2 />,
              <h3 />,
              <p />,
            ],
            "className": "information",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
